neo/rawio/intanrawio.py

```diff
@@ -530,6 +584,61 @@ def _demultiplex_digital_data(self, raw_digital_data, channel_ids, i_start, i_st
             output[:, channel_index] = demultiplex_data[i_start:i_stop].flatten()
 
         return output
+        
+    def _decode_current_from_stim_data(self, raw_stim_data, i_start, i_stop):
+        """
+        Demultiplex stimulation data by extracting current values from packed 16-bit format.
+        
+        According to the Intan RHS data format, stimulation current is stored in the lower 9 bits
+        of each 16-bit word: 8 bits for magnitude and 1 bit for sign. The upper bits contain
+        flags for compliance limit, charge recovery, and amplifier settle.
+        
+        Parameters
+        ----------
+        raw_stim_data : ndarray
+            Raw stimulation data in packed 16-bit format.
+        i_start : int
+            Starting sample index for demultiplexing.
+        i_stop : int
+            Ending sample index for demultiplexing (exclusive).
+            
+        Returns
+        -------
+        ndarray
+            Demultiplexed stimulation current values in amperes, preserving the original
+            array dimensions. The output values need to be multiplied by the stim_step_size
+            parameter (from header) to obtain the actual current in amperes.
+        
+        Notes
+        -----
+        Bit structure of each 16-bit stimulation word:
+        - Bits 0-7: Current magnitude
+        - Bit 8: Sign bit (1 = negative current)
+        - Bits 9-13: Unused (always zero)
+        - Bit 14: Amplifier settle flag (1 = activated)
+        - Bit 15: Charge recovery flag (1 = activated) 
+        - Bit 16 (MSB): Compliance limit flag (1 = limit reached)
+        
+        The actual current value in amperes is obtained by multiplying the
+        output by the 'stim_step_size' parameter from the file header.
```

Comment:
```suggestion
        output by the 'stim_step_size' parameter from the file header. These scaled values can be 
        obtained with the `rescale_signal_raw_to_float` function.
```

I know we deal with this at the SI level, but I think we should make this clear at the Neo level that the 
amps are only obtained if we rescale. So this requires two steps 1) get data which we now demultiplex 2) rescale the data which the user must do in a second function

Reply:
Sure.